File: Week4/VolvoCodingChallenge/app/src/main/java/com/yeah/ruisu/volvocodingchallenge/data/RepositoryModule/Repository.java
package com.yeah.ruisu.volvocodingchallenge.data.RepositoryModule;

import android.arch.lifecycle.MutableLiveData;

import com.yeah.ruisu.volvocodingchallenge.data.remote.models.ForecastDatum;

import java.util.List;

public interface Repository
{
    MutableLiveData<ForecastDatum> weatherForecastLiveData = null;

    void getWeatherForecast(String WOEID, String year, String month, String day);
}

File: week5/RealVolvoCodingChallenge/app/src/main/java/com/yeah/ruisu/realvolvocodingchallenge/data/RepositoryModule/Repository.kt
package com.yeah.ruisu.realvolvocodingchallenge.data.RepositoryModule

import android.arch.lifecycle.MutableLiveData
import com.yeah.ruisu.realvolvocodingchallenge.data.remote.models.ForecastDatum

interface Repository
{
    val weatherForecastLiveData : MutableLiveData<List<ForecastDatum>>

    fun getWeatherForecast(WOEID : String)
}